src/timetable_cmd.cpp

```diff
@@ -311,14 +311,19 @@ CommandCost CmdSetTimetableStart(DoCommandFlag flags, VehicleID veh_id, bool tim
 	CommandCost ret = CheckOwnership(v->owner);
 	if (ret.Failed()) return ret;
 
-	int total_duration = v->orders->GetTimetableTotalDuration();
+	TimerGameTick::Ticks total_duration = v->orders->GetTimetableTotalDuration();
 
-	/* Don't let a timetable start more than 15 years into the future or 1 year in the past. */
+	/* Don't let any timetables start at an invalid date. */
 	if (start_date < 0 || start_date > CalendarTime::MAX_DATE) return CMD_ERROR;
+	if (timetable_all && start_date + (total_duration / Ticks::DAY_TICKS) > CalendarTime::MAX_DATE) return CMD_ERROR;
+
+	/* Don't let a timetable start more than 15 years into the future... */
 	if (start_date - TimerGameCalendar::date > TimerGameCalendar::DateAtStartOfYear(MAX_TIMETABLE_START_YEARS)) return CMD_ERROR;
+	/* ...or 1 year in the past. */
 	if (TimerGameCalendar::date - start_date > CalendarTime::DAYS_IN_LEAP_YEAR) return CMD_ERROR;
+
+	/* A timetable must be complete. */
 	if (timetable_all && !v->orders->IsCompleteTimetable()) return CommandCost(STR_ERROR_TIMETABLE_INCOMPLETE);
-	if (timetable_all && start_date + total_duration / Ticks::DAY_TICKS > CalendarTime::MAX_DATE) return CMD_ERROR;
```

Comment:
This original ordering means that you will get an error about the timetable being incomplete before it checks the total duration. Does it make sense to check the total duration if it's incomplete?